src/imitation/scripts/train_adversarial.py

```diff
@@ -236,20 +177,136 @@ def should_plot_now(epoch) -> bool:
                                        trainer.env,
                                        n_episodes=n_episodes_eval)
     assert stats["n_traj"] >= n_episodes_eval
-    mean = stats["return_mean"]
-    std_err = stats["return_std"] / math.sqrt(n_episodes_eval)
-    print(f"[result] Mean Episode Return: {mean:.4g} ± {std_err:.3g} "
+    ep_reward_mean = stats["return_mean"]
+    ep_reward_std_err = stats["return_std"] / math.sqrt(n_episodes_eval)
+    print("[result] Mean Episode Return: "
+          f"{ep_reward_mean:.4g} ± {ep_reward_std_err:.3g} "
           f"(n={stats['n_traj']})")
 
-    return dict(mean=mean, std_err=std_err)
-
+    reward_path = os.path.join(log_dir, "checkpoints", "final", "discrim")
+    # TODO(shwang): I think Serializable should store the save_type, and
+    # Serializable.save() should return save_type.
+    if isinstance(trainer.discrim, DiscrimNetAIRL):
+      reward_type = "DiscrimNetAIRL"
+    elif isinstance(trainer.discrim, DiscrimNetGAIL):
+      reward_type = "DiscrimNetGAIL"
+    else:
+      raise RuntimeError(f"Unknown reward type for {trainer.discrim}")
+
+    return dict(ep_reward_mean=ep_reward_mean,
+                ep_reward_std_err=ep_reward_std_err,
+                log_dir=log_dir,
+                transfer_reward_path=reward_path,
+                transfer_reward_type=reward_type)
+
+
+class _TrainVisualizer:
+  def __init__(self,
+               trainer: "imitation.algorithms.adversarial.AdversarialTrainer",
```

Comment:
Why does this need to be in quotes? Can't we just import `AdversarialTrainer` (we're already importing the module it's contained in...)?